src/System.Globalization/tests/CultureInfo/CultureInfoAsync.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CultureInfoAsync
+    {        
+        [Fact]
+        public void TestCurrentCulturesAsync()
+        {
+            CultureInfo currentCulture = CultureInfo.CurrentCulture;
+            CultureInfo currentUICulture = CultureInfo.CurrentUICulture;
+
+            CultureInfo newCurrentCulture = new CultureInfo(currentCulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "en-US" : "ja-JP");
+            CultureInfo newCurrentUICulture = new CultureInfo(currentUICulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "en-US" : "ja-JP");
+            
+            CultureInfo.CurrentCulture = newCurrentCulture;
+            CultureInfo.CurrentUICulture = newCurrentUICulture;
+            
+            Task.Run(() => {
+                Assert.Equal(CultureInfo.CurrentCulture, newCurrentCulture);
+                Assert.Equal(CultureInfo.CurrentUICulture, newCurrentUICulture);
+            }).Wait();
```

Comment:
I'd suggest changing this to:
```C#
Task t = Task.Run() => {
    ...
});
((IAsyncResult)t).AsyncWaitHandle.WaitOne();
t.Wait();
```
The reason for that is Task.Wait tries to avoid blocking threads unnecessarily, and if you call Wait on a Task that hasn't yet started running, Wait may actually execute it on the current thread as part of the Wait call.  Normally that's a good thing for efficiency, but in this case you actually need the work to happen on a different thread in order to validate the semantics you're testing.